Directory.Build.props

```diff
@@ -3,13 +3,42 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <LangVersion>11.0</LangVersion>
     <Nullable>enable</Nullable>
-    <Features>strict</Features>
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+
+    <!--
+      Enable the latest warning wave, which shows additional warnings for invalid language features that are disabled by default.
+      For additional info, see https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/warning-waves.
+    -->
+    <AnalysisLevel>7</AnalysisLevel>
```

Comment:
You could try something like `7-recommended` or `7-all` which should import globalconfig from the CodeStyle package:

![image](https://user-images.githubusercontent.com/31348972/195933091-72d94559-2431-44a5-9b36-25aa56583415.png)

Reply:
That seems to work, but now I'm also getting a bazillion warnings for things that I actually don't want. If I want to override some of these defaults that are imported from the CodeStyle package, can I just do that in my .editorconfig?

Reply:
Yes. You can either enable every rule individually without using any globalconfig from the package, or use a globalconfig and start disabling what you don't need.

Reply:
Got it, that works! Question: what's the best way to disable a set of warnings in my "src\TerraFX.*" folders? Do I just add an .editorconfig in those folders that's not marked as root, and override the IDExxxx settings I want? 🤔

Rationale: those files are forked from another repo, so I don't want my code style to apply there as well.

Reply:
> Do I just add an .editorconfig in those folders that's not marked as root, and override the IDExxxx settings I want?

I was expecting this to work 😕

Reply:
Resolving this as we figured it out on Discord (missing `[*.cs]` headers) 😄